### PR TITLE
Use RSS for Walla

### LINKED ISSUE #1270: Walla - scrape news flash from RSS
Currently Walla is scraped from HTML, instead of https://rss.walla.co.il/feed/22
See discussion in #1261.

### PR DESCRIPTION
**Pull request by elazarg**
Fix #1270 
Sadly, it still requires parsing the HTML file in order to read the title (!). It's still useful though IMO.
Also, simplify the tests, reducing the number of different items.

Some obvious refactoring opportunities are left to the next step.

### CODE REVIEW

File: anyway/parsers/news_flash/beautiful_soup_news_flash_parse.py [hunk 1]
```diff
@@ -11,26 +10,31 @@
...
-    for item in news_items:
-        item_soup = item
-        entry_parsed_date = parsing_utils.get_date_time(item_soup, date, site_name)
-        if not ((latest_date is None) or (entry_parsed_date > latest_date)):
+    rss_soup = BeautifulSoup(response.text, "lxml")
+    news_items = parsing_utils.get_all_news_items(rss_soup, site_name)
+
+    for item_soup in news_items:
+        entry_parsed_date = parsing_utils.get_date_time(item_soup, site_name)
+        if latest_date is not None and entry_parsed_date <= latest_date:
```
Comment (elazarg) [inline anyway/parsers/news_flash/beautiful_soup_news_flash_parse.py:18]: Simple De Morgan

File: anyway/parsers/news_flash/beautiful_soup_news_flash_parse.py [hunk 2]
```diff
@@ -11,26 +10,31 @@
 def beautiful_soup_news_flash_parse(rss_link, site_name, maps_key):
     latest_date = get_latest_date_from_db(site_name)
     response = requests.get(rss_link)
-    html_soup = BeautifulSoup(response.text, "lxml")
-    news_items = parsing_utils.get_all_news_items(html_soup, site_name)
-    date = parsing_utils.get_date(html_soup, site_name)
```
Comment (elazarg) [inline anyway/parsers/news_flash/beautiful_soup_news_flash_parse.py:16]: This date is not needed anymore

File: anyway/parsers/news_flash/parsing_utils.py [hunk 3]
```diff
@@ -60,9 +60,9 @@ def get_all_news_items(html_soup, site_name):
     news_items = []
     try:
         if site_name == 'walla':
-            news_items = [s.find_parent('section') for s in html_soup.select('a[href^=/break/]')]
+            news_items = html_soup.find_all('item')
         elif site_name == 'ynet':
-            news_items = [i for i in html_soup.find_all('item') if i.category.get_text() == 'מבזקים']
```
Comment (elazarg) [inline anyway/parsers/news_flash/parsing_utils.py:65]: It is always מבזקים

**Comment by elazarg** [comment]
Embarrassingly, I forgot to actually change the link to RSS 🤦

**Comment by atalyaalon** [comment]
Create another pr lol

**Pull request merged**